### PR TITLE
Allow input_gradients to be set in constructor of NeuralNetwork

### LINKED ISSUE #90: Introduce a parameter `input_gradients` to NeuralNetwork
### What is the expected behavior?
As a feedback from #44,  we should add a parameter `input_gradients` to NeuralNetwork to expose it explicitly to users. By default it should be turned off. Also, we should not turn on input gradients explicitly in `TorchConnector` every time, turning on/off gradients is up a user via this new parameter.

### PR DESCRIPTION
**Pull request by jonvet**
### Summary
This PR introduces a parameter ``input_gradients`` to the constructor of ``NeuralNetwork`` and its subclasses. ``input_gradients`` defaults to False. The constructor of ``TorchConnector`` no longer sets ``input_gradients=True`` by default on the ``NeuralNetwork`` it was instantiated with. This PR closes #90.

### Details and comments
 If you are using ``TorchConnector`` and want to compute gradients w.r.t. to inputs, you will have to set ``input_gradients`` explicitly on the ``NeuralNetwork`` you pass to ``TorchConnector``.

Changelog: New Feature

### CODE REVIEW

File: qiskit_machine_learning/algorithms/classifiers/vqc.py
Comment (stefan-woerner) [inline qiskit_machine_learning/algorithms/classifiers/vqc.py:53]: This can be just set to `False` for VQC. Input gradients make only sense if the QNN is embedded in a larger workflow, VQC has no way to use it (otherwise one could use the NNClassifier and construct it manually).

File: qiskit_machine_learning/algorithms/classifiers/vqc.py
Comment (stefan-woerner) [inline qiskit_machine_learning/algorithms/classifiers/vqc.py:114]: see previous comment, can just be set to `False`.

**Review by stefan-woerner — CHANGES_REQUESTED** [review]
Thanks! Could you have a look and update the VQC (and accordingly also the VQR which is not included in this PR yet).

**Comment by jonvet** [comment]
all tests pass locally, but some fail on the CI. In particular, in the last build `test_classifier_with_opflow_qnn_4___cobyla____l2____qasm__` failed and yesterday `test_regressor_with_opflow_qnn_3___bfgs____statevector__` failed - but both pass on my laptop. any tips on how to debug this?

**Comment by adekusar-drl** [comment]
> 
> 
> all tests pass locally, but some fail on the CI. In particular, in the last build `test_classifier_with_opflow_qnn_4___cobyla____l2____qasm__` failed and yesterday `test_regressor_with_opflow_qnn_3___bfgs____statevector__` failed - but both pass on my laptop. any tips on how to debug this?

This is a common problem, should be fixed separately, it is not a problem in your code. Thanks for the PR!

**Comment by adekusar-drl** [comment]
@stefan-woerner please review again, the merge is blocked until your approval.

**Review by stefan-woerner — APPROVED** [review]
lgtm

**Comment by adekusar-drl** [comment]
@jonvet Thanks a lot for your contribution! Really appreciated!

**Pull request merged**
